app/controllers/api/v1/items/search_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Api::V1::Items::SearchController < ApplicationController
+
+  def show
+    query_keys = params.keys[0..-3]
+    query_values = []
+    query_keys.each { |key| query_values << params[key] }
+    query_keys.zip(query_values)
+
```

Comment:
this is not working, but the controller and method name are good foundation. May need to use a poro